src/sass/patterns/_containers.scss

```diff
@@ -48,3 +48,10 @@
     height: 100%;
   }
 }
+
+// Remove extra space padding by overriding slds-p-right_medium on narrow screens
+@media (max-width: $mq-medium) {
+  .search-container {
```

Comment:
@dvdherron great idea. I wish LDS had some way to set classes for certain breakpoints like they do for columns.